android/app/src/bt/java/covidsafepaths/bt/exposurenotifications/storage/RealmSecureStorageBte.kt

```diff
@@ -118,6 +120,33 @@ object RealmSecureStorageBte {
         }
     }
 
+    fun refreshWithExposureWindows(exposureWindows: List<ExposureWindow>): Boolean {
+        var somethingAdded = false
+        getRealmInstance().use {
+            it.executeTransaction { db ->
+                // Keep track of the exposures already handled and remove them when we find matching windows.
+                val exposureEntities: MutableList<ExposureEntity> = db.where(ExposureEntity::class.java).findAll()
```

Comment:
Can you explain more what is going on here?

Reply:
I got this logic from here https://github.com/google/exposure-notifications-android/blob/master/app/src/main/java/com/google/android/apps/exposurenotification/storage/ExposureDao.java and adapted it to use `Realm` instead of `Room`.

We save in the database the exposures that were already handled to avoid showing the notification again.
If we get a new exposure then we save it in the database, this method returns true and we display the notification.